src/error.rs

```diff
@@ -30,6 +30,13 @@ pub(crate) enum Error {
   FileSearch { source: ignore::Error },
   #[snafu(display("Invalid glob: {}", source))]
   GlobParse { source: globset::Error },
+  #[snafu(display("Failed to parse hex string `{}`: {}", text, source))]
```

Comment:
I try to avoid variants with message fields that explain why the error is occuring for normal errors. Instead, I would add a separate error type, `MagnetLinkParseError`, and create variants for all the different ways that magnet link parsing could fail, and then an `Error::MagnetLinkParse` variant that contains the `MagnetLinkParseError`.